runtime-common/search-index.ts

```diff
@@ -682,6 +723,36 @@ export class SearchIndex {
         });
     }
 
+    if ("range" in filter) {
+      let ref: CardRef = { type: "exportedCard", ...on };
+
+      await Promise.all(
+        Object.keys(filter.range).map((fieldPath) =>
+          this.validateField(ref, fieldPath.split("."))
+        )
+      );
+
+      return (entry) =>
+        every(Object.entries(filter.range), ([fieldPath, range]) => {
+          if (this.cardHasType(entry, ref)) {
+            let value = entry.searchData[fieldPath];
+            if (value === undefined) {
+              return null;
```

Comment:
I think this is correct as the least surprising behavior for undefined values.

But then for consistency I think we should change `eq` to also work this way. Fields with value of undefined would always result in `null` instead of `false` so they can't be inverted by a `not` filter into matching the search.